Wikipedia/Code/ShiftingNavigationBarView.swift

```diff
@@ -0,0 +1,117 @@
+import UIKit
+
+class ShiftingNavigationBarView: ShiftingTopView, Themeable {
+
+    private let navigationItems: [UINavigationItem]
+    private weak var popDelegate: UIViewController? // Navigation back actions will be forwarded to this view controller
+    
+    private var topConstraint: NSLayoutConstraint?
+
+    private lazy var bar: UINavigationBar = {
+        let bar = UINavigationBar()
+        bar.translatesAutoresizingMaskIntoConstraints = false
+        bar.delegate = self
+        return bar
+    }()
+
+    init(shiftOrder: Int, navigationItems: [UINavigationItem], popDelegate: UIViewController) {
+        self.navigationItems = navigationItems
+        self.popDelegate = popDelegate
+        super.init(shiftOrder: shiftOrder)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func setup() {
+        super.setup()
+
+        bar.setItems(navigationItems, animated: false)
+        addSubview(bar)
+
+        let top = bar.topAnchor.constraint(equalTo: topAnchor)
+        let bottom = bottomAnchor.constraint(equalTo: bar.bottomAnchor)
+        let leading = bar.leadingAnchor.constraint(equalTo: leadingAnchor)
+        let trailing = trailingAnchor.constraint(equalTo: bar.trailingAnchor)
+
+        NSLayoutConstraint.activate([
+            top,
+            bottom,
+            leading,
+            trailing
+        ])
+
+        self.topConstraint = top
+
+        clipsToBounds = true
+    }
+
+    // MARK: Overrides
+    
+    override var contentHeight: CGFloat {
+        return bar.frame.height
+    }
+
+    private var isFullyHidden: Bool {
+       return -(topConstraint?.constant ?? 0) == contentHeight
+    }
+
+    override func shift(amount: CGFloat) -> ShiftingTopView.AmountShifted {
+
+        // Only allow navigation bar to move just out of frame
```

Comment:
It would be good to prevent the bar from hiding when VoiceOver is on, or provide another way for VoiceOver users to go back from this view when it is hiding (This can be done on the part 2 PR).